apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/Keys.cs

```diff
@@ -101,10 +98,39 @@ private KeySet(bool allKeys, IEnumerable<Key> keys, IEnumerable<KeyRange> ranges
 
         /// <inheritdoc/>
         public override bool Equals(object other)
```

Comment:
Given that these are now mutable, do we really need to implement Equality?
I would rather not I think, @jskeet thoughts?

Reply:
We should probably find out why they implemented equality in the first place - probably just for convenience, knowing it was safe as they were immutable and based on protos.

If we want to make a breaking change by removing the equality, now is the time to do so.

Reply:
As far as I can tell, it was just for convenience and not really needed. The Read API is not used in any of the ORM implementations that we have.